### PR TITLE
Bug1195653 - Part2: Show mask effect on LayerScope viewer.

### PR DESCRIPTION
**Pull request by chenpighead**
Bug 1195653 - part2.1: Show isMask attribute on viewer title bar.
Bug 1195653 - Part2.2: Let sprites in texture view be displayed in order of layerID, so mask effect can be shown right next to its primary effect.

### CODE REVIEW

File: js/renderer/texturesprite_renderer.js [hunk 1]
```diff
@@ -59,7 +66,31 @@ LayerScope.SpriteView = {
...
+    if (index == -1) {
+      this._sortedLayerIDs.push(layerID);
+      this._sortedSprites.push($sprite);
+    } else {
+      this._sortedLayerIDs.splice(index, 0, layerID);
+      this._sortedSprites.splice(index, 0, $sprite);
+    }
+  },
+
+  _createTexSprites: function SV_createTexSprites(frame, $panel, sortedSprites) {
```
Comment (CJKu) [inline js/renderer/texturesprite_renderer.js]: sortedSprites is a parameter never been used.

File: js/renderer/texturesprite_renderer.js [hunk 2]
```diff
@@ -49,9 +51,14 @@ LayerScope.SpriteView = {
...
+      this._sortedLayerIDs = [];
+      this._sortedSprites = [];
 
-      this._createTexSprites(frame, this._$panel);
-      this._createColorSprites(frame, this._$panel);
+      this._createTexSprites(frame, this._$panel, this._sortedSprites);
+      this._createColorSprites(frame, this._$panel, this._sortedSprites);
+      for (let i = 0; i < this._sortedSprites.length; i++) {
+        this._$panel.append(this._sortedSprites[i]);
+      }
```
Comment (CJKu) [inline js/renderer/texturesprite_renderer.js]: You may use this._sortedSprites.forEach here

File: js/renderer/texturesprite_renderer.js [hunk 3]
```diff
@@ -59,7 +66,31 @@ LayerScope.SpriteView = {
     this._resizeSprites(this._textures, this._colors);
   },
 
-  _createTexSprites: function SV_createTexSprites(frame, $panel) {
+  _sortSpriteByLayerID: function SV_sortSpriteByLayerID(layerID, $sprite) {
+    if (this._sortedLayerIDs === [] || this._sortedSprites === []) {
+      this._sortedLayerIDs.push(layerID);
+      this._sortedSprites.push($sprite);
```
Comment (CJKu) [inline js/renderer/texturesprite_renderer.js]: You may carry layerID in $sprite and eliminate this._sortedLayerIDs

File: js/renderer/texturesprite_renderer.js [hunk 4]
```diff
@@ -59,7 +66,31 @@ LayerScope.SpriteView = {
...
+      return;
+    }
+
+    var index = -1;
+    for (i = 0; i < this._sortedLayerIDs.length; i++) {
+      if (layerID < this._sortedLayerIDs[i]) {
+        index = i;
+        break;
```
Comment (CJKu) [inline js/renderer/texturesprite_renderer.js]: this._sortedLayerIDs.splice(index, 0, layerID);
  this._sortedSprites.splice(index, 0, $sprite);
  return;

File: js/renderer/texturesprite_renderer.js [hunk 5]
```diff
@@ -59,7 +66,31 @@ LayerScope.SpriteView = {
...
+      }
+    }
+
+    if (index == -1) {
+      this._sortedLayerIDs.push(layerID);
+      this._sortedSprites.push($sprite);
+    } else {
```
Comment (CJKu) [inline js/renderer/texturesprite_renderer.js]: remove this else statement

File: js/renderer/texturesprite_renderer.js [hunk 6]
```diff
@@ -59,7 +66,31 @@ LayerScope.SpriteView = {
...
+
+    var index = -1;
+    for (i = 0; i < this._sortedLayerIDs.length; i++) {
+      if (layerID < this._sortedLayerIDs[i]) {
+        index = i;
+        break;
+      }
+    }
+
+    if (index == -1) {
```
Comment (CJKu) [inline js/renderer/texturesprite_renderer.js]: You don't need if expression here. add spite into sorted list directly.

**Comment by BorisChiou** [comment]
LGTM

**Pull request merged**